great_expectations/data_context/data_context/file_data_context.py

```diff
@@ -19,20 +20,18 @@
 logger = logging.getLogger(__name__)
 
 
-class FileDataContext(AbstractDataContext):
+class FileDataContext(SerializableDataContext):
     """
     Extends AbstractDataContext, contains only functionality necessary to hydrate state from disk.
 
     TODO: Most of the functionality in DataContext will be refactored into this class, and the current DataContext
     class will exist only for backwards-compatibility reasons.
     """
 
-    GX_YML = "great_expectations.yml"
-
     def __init__(
         self,
-        project_config: DataContextConfig,
         context_root_dir: str,
+        project_config: Optional[DataContextConfig] = None,
```

Comment:
If not present, we should have legacy `DataContext` behavior where we parse it from the filesystem